module/Calculator/SpeedCalculator.cpp

```diff
@@ -8,33 +8,58 @@
 SpeedCalculator::SpeedCalculator(double _targetSpeed)
   : targetSpeed(_targetSpeed), pid(0.001, 0.000000001, 0.0001, _targetSpeed)
 {
-  pwm = 0.0;
+  rightPwm = 0.0;
+  leftPwm = 0.0;
   int rightAngle = Measurer::getRightCount();
   int leftAngle = Measurer::getLeftCount();
-  prevMileage = Mileage::calculateMileage(rightAngle, leftAngle);
-  prevTime = timer.now();
+  prevRightMileage = Mileage::calculateWheelMileage(rightAngle);
+  prevLeftMileage = Mileage::calculateWheelMileage(leftAngle);
+  prevRightTime = timer.now();
+  prevLeftTime = timer.now();
 }
 
-int SpeedCalculator::calcPwmFromSpeed()
+int SpeedCalculator::calcRightPwmFromSpeed()
 {
-  // 左右タイヤの回転角度を取得
+  // 右タイヤの回転角度を取得
   int rightAngle = Measurer::getRightCount();
+  // 右タイヤの走行距離を算出
+  double currentRightMileage = Mileage::calculateWheelMileage(rightAngle);
+  double diffRightMileage = currentRightMileage - prevRightMileage;
+  // 走行時間を算出
+  int currentRightTime = timer.now();
+  double diffRightTime = (double)(currentRightTime - prevRightTime);
+  // 右タイヤの走行速度を算出
+  double currentRightSpeed
+      = (diffRightMileage == 0.0) ? -targetSpeed : calcSpeed(diffRightMileage, diffRightTime);
```

Comment:
よくよく考えるとこっちの方が調整しやすいかも。
(もしかしたら、テストもこの処理で初速が上がりすぎてる可能性はある)

```suggestion
  double currentRightSpeed
      = calcSpeed(diffRightMileage, diffRightTime);
```